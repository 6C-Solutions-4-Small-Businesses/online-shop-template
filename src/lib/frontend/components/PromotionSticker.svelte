<script lang="ts">
    export let percentage: number

    function getDiscountClass(): string {
        if (percentage <= 10) {
            return 'bg-promotion-low';
        } else if (percentage <= 50) {
            return 'bg-promotion-medium';
        } else {
            return 'bg-promotion-high';
        }
    }
</script>

<div class={`absolute left-0 flex w-12 h-12 top-0 items-center justify-center rounded-tl-[10px] rounded-tr-none rounded-br-[20px] rounded-bl-none ${getDiscountClass()} ${percentage ? 'visible z-10' : 'invisible'}`}>
    <div class="text-white text-xs font-bold">
        -{percentage}%
    </div>
</div>