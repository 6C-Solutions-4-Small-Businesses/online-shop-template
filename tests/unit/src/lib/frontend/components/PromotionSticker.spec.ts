import {describe} from 'vitest'
import PromotionSticker from '$lib/frontend/components/PromotionSticker.svelte'
import {render} from '@testing-library/svelte'

describe('Promotion sticker', () => {
    it('should have match snapshot', () => {
        const {container} = render(PromotionSticker, {
            percentage: 85
        })

        expect(container).toMatchSnapshot()
    })
})