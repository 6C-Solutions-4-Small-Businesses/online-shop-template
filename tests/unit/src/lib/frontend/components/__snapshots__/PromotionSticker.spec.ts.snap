// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Promotion sticker > should have match snapshot 1`] = `
<div>
  <div
    class="absolute left-0 flex w-12 h-12 top-0 items-center justify-center rounded-tl-[10px] rounded-tr-none rounded-br-[20px] rounded-bl-none bg-promotion-high visible z-10"
  >
    <div
      class="text-white text-xs font-bold"
    >
      -
      85
      %
    </div>
  </div>
</div>
`;
